src/Symfony/Component/Routing/Loader/LoaderHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+/**
+ * Common parsing logic between route loaders.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+trait LoaderHelper
+{
+    private function redirectToDefaults(string $redirect, bool $permanent = false): array
+    {
+        if ($this->looksLikeUrl($redirect)) {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction';
```

Comment:
would be better to write it as `$defaults = array(..., ...)` instead of assigning in an undefined variable

Reply:
Not sure how that happened - well, I know, refactoring.. - , I'd expect PHP to tell me how bad was that.